### PR TITLE
Fix ValidateRange Max Depth Allowed

### LINKED ISSUE #15344: ConvertTo-Json allows Depth value of int.MaxValue but only supports maximum of 100
See https://github.com/PowerShell/PowerShell/blob/a95b2cb17fecc8640533420b3baa9aec0f3749cb/src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/ConvertToJsonCommand.cs#L34-L43

```csharp
        /// <summary>
        /// Gets or sets the Depth property.
        /// </summary>
        [Parameter]
        [ValidateRange(0, int.MaxValue)]
        public int Depth
        {
            get { return _depth; }
            set { _depth = value; }
        }
```

The valid range should be 0 to 100, not int.MaxValue.

## Steps to reproduce

```powershell
PS C:\Git> dir | ConvertTo-Json -Depth 1000
ConvertTo-Json: The maximum depth allowed for serialization is 100.
```

## Expected behavior

The error should come from parameter validation, not from `BeginProcessing()`.

## Environment data

```none
Name                           Value
----                           -----
PSVersion                      7.1.3
PSEdition                      Core
GitCommitId                    7.1.3
OS                             Microsoft Windows 10.0.19043
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0ΓÇª}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

### PR DESCRIPTION
**Pull request by KevRitchie**
# PR Summary

Replace int.MaxValue with 100 so that parameter validation works as expected.

Remove BeginProcessing method.  No longer required to do a prerequisite check on Max Depth.  Parameter Argument Validation now used.

Update Json Tests.

## PR Context

Fixes #15344

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [] N/A or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and opened an issue in the relevant tool repository. This may include:
        - [ ] Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode
        (which runs in a different PS Host).
            - [ ] Issue filed: 
        - [ ] Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
            - [ ] Issue filed: 
        - [ ] Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
            - [ ] Issue filed: 
        - [ ] Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).
            - [ ] Issue filed:

### CODE REVIEW

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/ConvertToJsonCommand.cs [hunk 1]
```diff
@@ -99,23 +99,7 @@ protected virtual void Dispose(bool disposing)
                 _cancellationSource.Dispose();
             }
         }
-
-        /// <summary>
-        /// Prerequisite checks.
-        /// </summary>
-        protected override void BeginProcessing()
-        {
-            if (_depth > maxDepthAllowed)
```
Comment (daxian-dbw) [inline src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/ConvertToJsonCommand.cs:108]: The variable `maxDepthAllowed` should be removed too.

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/ConvertToJsonCommand.cs [hunk 2]
```diff
@@ -99,23 +99,7 @@ protected virtual void Dispose(bool disposing)
...
-            {
-                string errorMessage = StringUtil.Format(WebCmdletStrings.ReachedMaximumDepthAllowed, maxDepthAllowed);
```
Comment (daxian-dbw) [inline src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/ConvertToJsonCommand.cs:110]: The resource string `ReachedMaximumDepthAllowed` should be removed. It's only used in `ConvertTo-Json`.
The resource string can be found in `src\Microsoft.PowerShell.Commands.Utility\resources\WebCmdletStrings.resx`

**Comment by daxian-dbw** [comment]
@SteveL-MSFT can you please take a quick look?

**Comment by iSazonov** [comment]
I wonder why do we change the check if we get right error message? Also see #3181 - I wonder about it too.

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/ConvertToJsonCommand.cs (hunk 1 above)
Comment (KevRitchie) [reply src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/ConvertToJsonCommand.cs:108]: maxDepthAllowed removed.

**Review by daxian-dbw — APPROVED** [review]

**Comment by daxian-dbw** [comment]
The documentation is already up-to-date;
> Specifies how many levels of contained objects are included in the JSON representation. The value can be any number from 1 to 100. The default value is 2. ConvertTo-Json emits a warning if the number of levels in an input object exceeds this number.

@iSazonov This is to fix the incorrect validate range attribute. While after fixing that, there is no need for explicit check.

**Comment by daxian-dbw** [comment]
> The documentation is already up-to-date;

Actually, there is an inconsistency in the doc. `-Depth` supports argument 0, but the help content say `from 1 to 100`.
@sdwheeler, shall we update the doc?

**Pull request merged**